app/services/usps_in_person_proofing/enrollment_helper.rb

```diff
@@ -39,23 +44,53 @@ def usps_proofer
     end
 
     def create_usps_enrollment(enrollment, pii)
+      address = pii['address1']
+      address += " #{pii['address2']}" unless pii['address2'].blank?
+
       applicant = UspsInPersonProofing::Applicant.new(
         {
           unique_id: enrollment.usps_unique_id,
           first_name: pii['first_name'],
           last_name: pii['last_name'],
-          address: pii['address1'],
-          # do we need address2?
+          address: address,
           city: pii['city'],
           state: pii['state'],
           zip_code: pii['zipcode'],
           email: 'no-reply@login.gov',
         },
       )
+
       proofer = usps_proofer
+      response = nil
+      begin
+        response = proofer.request_enroll(applicant)
+      rescue Faraday::BadRequestError => err
+        handle_bad_request_error(err, enrollment)
+      rescue StandardError => err
+        handle_standard_error(err, enrollment)
+      end
 
-      response = proofer.request_enroll(applicant)
-      response['enrollmentCode']
+      response&.enrollmentCode
+    end
+
+    def handle_bad_request_error(err, enrollment)
+      analytics.idv_in_person_usps_request_enroll_exception(
+        context: context,
+        enrollment_id: enrollment.id,
+        exception_class: err.class.to_s,
+        exception_message: err.response.dig(:body, 'responseMessage') || err.message,
+        reason: 'Request exception',
+      )
+    end
+
+    def handle_standard_error(err, enrollment)
+      analytics.idv_in_person_usps_request_enroll_exception(
+        context: context,
+        enrollment_id: enrollment.id,
+        exception_class: err.class.to_s,
+        exception_message: err.message,
+        reason: 'Request exception',
+      )
```

Comment:
Couple thoughts:

- We could make the `rescue`s method level and avoid the `response` variable initialization and nil-safe operation
- Since the error handling logic here is largely identical, and since there's already precedent for wrapping the errors with `StandardError`, and since ideally awareness of Faraday implementation details is limited to the `Proofer` class where it is called, could we move the Faraday rescue-ing into that class?

```rb
# enrollment_helper.rb

    def create_usps_enrollment(enrollment, pii)
      # ...

      proofer = usps_proofer
      response = proofer.request_enroll(applicant)
      response.enrollmentCode
    rescue StandardError => error
      handle_error(error, enrollment)
    end

    def handle_error(error, enrollment)
      analytics.idv_in_person_usps_request_enroll_exception(
        context: context,
        enrollment_id: enrollment.id,
        exception_class: error.class.to_s,
        exception_message: error.message,
        reason: 'Request exception',
      )
    end
```


```rb
# proofer.rb

    def request_enroll(applicant)
      # ...

      res = faraday.post(url, body, dynamic_headers).body
      Response::RequestEnrollResponse.new(res)
    rescue Faraday::BadRequestError => error
      raise StandardError.new(error.response.dig(:body, 'responseMessage') || error.message)
    end
```

Reply:
Cool, yeah, I've simplified the rescue logic in [244c456](https://github.com/18F/identity-idp/pull/6629/commits/244c456edd97a7dbb6de24464738c5b04028879a)

I ended up adding a new `UspsInPersonProofing::Exception::RequestEnrollException` wrapper class for errors that occur -- this allows them to be rescued by the controller for logging analytics. Any concerns with that instead of just throwing StandardErrors?

Reply:
Yeah I think that works 👍 

_Maybe_ it makes the controllers a bit too aware of the implementation details of the enrollment (vs. rescuing it closer to the point where we're calling `schedule_in_person_enrollment`), but in this case I think it's the most convenient way to handle the analytics and response. In other controllers, I could see this being the sort of thing handled by a `FormResponse` class, where the exception message would be tracked as an error, and the controller would have common response logic based on `!response.success?` ([example](https://github.com/18F/identity-idp/blob/ecc6d330eea56465e10b6feabc756d8e727e9c09/app/forms/new_phone_form.rb#L98-L114)).